core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionDetails.java

```diff
@@ -69,5 +72,17 @@ default String getRootPath() {
    *
    * @param rootPath The root path
    */
-  default void setRootPath(String rootPath ) {}
+  default void setRootPath( String rootPath ) { }
+
+  /**
+   * Gets the list of roles with access to the connection from the Business Analytics product.
+   * <p>
+   * Access control does not distinguish between types of access operations (such as read, write, delete).
+   * Access is granted to either all or none of the operations.
+   * @return A non-null list of roles.
+   */
+  default List<String> getBaRoles() {
```

Comment:
the point of this comment is to note my concerns. **No action for now.**
I'm not a stickler on naming However in this case

 "BA" or "business analytics" in kettle code might be confusing since Kettle code is shared between PDI, PUC, and other products.
 
 There was mention of topic in slack https://hitachivantara-eng.slack.com/archives/C01HJ874L8Z/p1713382434073959 which i need to review.

Reply:
Keeping just the field, getter and setter for the BA-specific field is the minimum impact solution we could find, while moving any BA-specific business logic to the BA connection manager service, as advised by @abryant-hv.

Naming this field, getter and setter generically, such as in`roles`, would give the false impression that access control should be honored by PDI as well.

The only other solution I can think of would be to have a generic map of properties which could hold arbitrary properties for specific "clients".